src/Stack/Setup.hs

```diff
@@ -1226,7 +1227,23 @@ getGhcBuilds = do
     config <- view configL
     case configGHCBuild config of
         Just ghcBuild -> pure [ghcBuild]
-        Nothing -> determineGhcBuild
+        Nothing -> do
+          mOsReleaseResult <- liftIO OsRelease.parseOsRelease
+          case mOsReleaseResult of
+            Nothing -> determineGhcBuild
+            Just OsRelease.OsReleaseResult{OsRelease.osRelease=osRelease} -> do
+              let OsRelease.OsRelease{OsRelease.id=osId, OsRelease.id_like=osIdLike} = osRelease
+              -- The heuristic in determineGhcBuild chooses a Fedora build for
+              -- GHC 9.4 on Debian 11 and Ubuntu 20.04, which doesn't work.
+              -- Make sure we use the standard build on Debian-like distributions.
+              -- Up until now, it has always been a Debian build.
+              -- Note that we don't verify the operating system version, so it
+              -- could still be wrong.
+              if elem (fromMaybe osId osIdLike) ["debian", "ubuntu"] then
```

Comment:
You probably want:

I'd include `Name` field and then

1. `toLower` and
2. detect the *word* Debian, such that `Debian GNU/Linux` or `GNU/Linux Debian` matches.

Because neither Name, nor ID, nor ID_LIKE are guaranteed to exist and the spec defines defaults.

See here: https://github.com/haskell/ghcup-hs/blob/a5f2067d76fb529d79ae28cca1bb11ed15baa4d0/lib/GHCup/Platform.hs#L124

Reply:
There's also `lsb-release` and the debian version file. GHCup deals with all of them.

Reply:
Thanks for the quick feedback.

> There's also lsb-release

On my Debian 11 machine, the executable is called `lsb_release`, with an underscore, not a hyphen. I am a bit puzzled as to why GHCup has `lsb_release_cmd = "lsb-release"`. But I don't think the fallback should be necessary.

> toLower

The spec defines ID and ID_LIKE to be lower case. But I have added a comment to call `toLower` anyway as suggested.

> detect the word Debian

I added a call to `words` for ID_LIKE, which the standard indeed suggests could contain spaces.

> neither Name, nor ID, nor ID_LIKE are guaranteed to exist

The Debian derivatives I have tested, Devuan, Mint and Ubuntu, all have either the ID or the ID_LIKE fields (see the gist linked in the issue). The bug has only been observed on Ubuntu 20.04 and Debian 11, so if the heuristic fails on other versions, we'd just be falling back to existing behavior, which I think is fine, since those derivatives/versions are very rare, and it would still be possible to specify the ghc-build manually as is currently needed on Debian 11/Ubuntu 20.04.

If you insist, I can add the name fallback as suggested. It just seems unnecessarily complex, am I missing something?

> the spec defines defaults.

In the case that `ID=linux` as the spec mentions as the default, we'd use existing logic, which is fine! Because the user is on a different distribution then, or a very old Debian derivative.